core/src/mast/mod.rs

```diff
@@ -76,6 +79,10 @@ pub struct MastForest {
     /// Provides efficient access to decorators per operation per node during execution and
     /// debugging.
     decorator_storage: DecoratorIndexMapping,
+
+    /// Storage for node-level decorators (before_enter and after_exit). This uses CSR format
+    /// for efficient storage and access of node-level decorators.
+    node_decorator_storage: NodeDecoratorStorage,
```

Comment:
IIUC,
- `DecoratorIndexMapping`: only relevant for operations *within* basic blocks (i.e. no before_enter/after_exit),
- `NodeDecoratorStorage`: the `before_enter`/`after_exit` decorators for all node types.

Therefore I would rename the following:
- `decorator_storage` -> `op_decorator_storage`
- `DecoratorIndexMapping` -> `OpToDecoratorIdsMapping` or `OpToDecoratorIds`
- `NodeDecoratorStorage` -> `NodeToDecoratorIdsMapping` or `NodeToDecoratorIds`


Or any other naming schemes that make that relationship more explicit.